From 870909e6ae20b9f1b6402da60eeb21762864b191 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Wed, 29 Apr 2020 08:03:01 -0500
Subject: [PATCH 23/25] arm64: dts: Fix UART and SPI Bus on Beta Hardware

The SPI bus uses one of the UART pins which make it unable to do
harware handshaking.

This patch, moves the SPI2 SS0 pin to ecspi2 which frees up the hardware
handshaking on UART3.

Fixes: PHOENIX8-53

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../freescale/beacon-imx8mm-baseboard.dtsi    |  4 +++-
 .../dts/freescale/beacon-imx8mm-kit-alpha.dts | 22 +++++++++++++++++++
 2 files changed, 25 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
index 48bba64db8e6..d84c4d758f75 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
@@ -194,7 +194,7 @@
 				MX8MM_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK		0x82
 				MX8MM_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI		0x82
 				MX8MM_IOMUXC_ECSPI2_MISO_ECSPI2_MISO		0x82
-				MX8MM_IOMUXC_ECSPI1_SS0_GPIO5_IO9		0x41
+				MX8MM_IOMUXC_ECSPI2_SS0_ECSPI2_SS0		0x82
 			>;
 		};
 
@@ -265,6 +265,8 @@
 			fsl,pins = <
 				MX8MM_IOMUXC_ECSPI1_SCLK_UART3_DCE_RX		0x140
 				MX8MM_IOMUXC_ECSPI1_MOSI_UART3_DCE_TX		0x140
+				MX8MM_IOMUXC_ECSPI1_MISO_UART3_DCE_CTS_B	0x140
+				MX8MM_IOMUXC_ECSPI1_SS0_UART3_DCE_RTS_B		0x140
 			>;
 		};
 
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-kit-alpha.dts b/arch/arm64/boot/dts/freescale/beacon-imx8mm-kit-alpha.dts
index 765006facb44..954a3c44f684 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-kit-alpha.dts
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-kit-alpha.dts
@@ -25,4 +25,26 @@
 	/delete-property/ enable-active-high;
 };
 
+/delete-node/ &pinctrl_espi2;
+/delete-node/ &pinctrl_uart3;
+
+&iomuxc {
+	/* Alpha hardware uses a different SS0 pin than beta */
+	pinctrl_espi2: espi2grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_ECSPI2_SCLK_ECSPI2_SCLK		0x82
+			MX8MM_IOMUXC_ECSPI2_MOSI_ECSPI2_MOSI		0x82
+			MX8MM_IOMUXC_ECSPI2_MISO_ECSPI2_MISO		0x82
+			MX8MM_IOMUXC_ECSPI1_SS0_GPIO5_IO9		0x41
+		>;
+	};
+
+	/* UART 3 on Beta has no hw flow control */
+	pinctrl_uart3: uart3grp {
+		fsl,pins = <
+			MX8MM_IOMUXC_ECSPI1_SCLK_UART3_DCE_RX		0x140
+			MX8MM_IOMUXC_ECSPI1_MOSI_UART3_DCE_TX		0x140
+		>;
+	};
+};
 
-- 
2.17.1

